collector/multiplier.go

```diff
@@ -0,0 +1,72 @@
+package collector
+
+import (
+	"context"
+	"sync"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Config struct {
+	Collectors []Interface
+	Logger     micrologger.Logger
+}
+
+type Multiplier struct {
+	collectors []Interface
+	logger     micrologger.Logger
+
+	bootOnce sync.Once
+}
+
+func New(config Config) (*Multiplier, error) {
+	if len(config.Collectors) == 0 {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Collectors must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	m := &Multiplier{
+		collectors: config.Collectors,
+		logger:     config.Logger,
+
+		bootOnce: sync.Once{},
+	}
+
+	return m, nil
+}
+
+func (m *Multiplier) Boot(ctx context.Context) {
+	m.logger.Log("level", "debug", "message", "booting collector")
+
+	m.bootOnce.Do(func() {
+		for _, c := range m.collectors {
+			c.Boot(ctx)
+		}
+	})
+
+	m.logger.Log("level", "debug", "message", "booted collector")
+}
+
+func (m *Multiplier) Collect(ch chan<- prometheus.Metric) {
+	m.logger.Log("level", "debug", "message", "collecting metrics")
+
+	for _, c := range m.collectors {
+		c.Collect(ch)
```

Comment:
I think it would be nice to `recover` here to make sure our collectors don't kill the operator.